packages/webdriver-utils/src/providers/automateProvider.js

```diff
@@ -83,12 +83,25 @@ export default class AutomateProvider extends GenericProvider {
           percyBuildUrl: this.buildInfo.url,
           state: 'begin'
         });
+        // Selenium Hub, set status error Code to 13 if an error is thrown
+        // Handling error with Selenium dialect is != W3C
+        if (result?.status === 13) throw new Error(result?.value || 'Got invalid error response');
         this._markedPercy = result.success;
         return result;
       } catch (e) {
         log.debug(`[${name}] : Could not mark Automate session as percy`);
         log.error(`[${name}] : error: ${e.toString()}`);
-        return null;
+        /**
+         * - Handling Error when dialect is W3C
+         * ERROR response format from SeleniumHUB `{
+         * sessionId: ...,
+         * status: 13,
+         * value: { error: '', message: ''}
+         * }
+         */
+        const errResponse = (e?.response?.body && JSON.parse(e?.response?.body)?.value) || {};
+        const errMessage = errResponse?.message || errResponse?.error || e?.message || e?.error || e?.value || e.toString();
```

Comment:
Why so many `OR`'s ?

Reply:
we are handling 2 cases 
1. for failed response
2. If the err is thrown from the try block
Generally, error comes in e.message, or e.error or e.value, but if err doesn't comes in these e.toString() is shown.